File: 0704-binary-search/0704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        return Binarysearch(nums, target, 0, nums.length - 1);
    }
    
    int Binarysearch(int[] arr, int target, int s, int e){
        
     if (s > e) {
            return -1;
        }
        int m = s + (e - s) / 2;
        if (arr[m] == target) {
            return m;
        }
        if (target < arr[m]) {
            return Binarysearch(arr, target, s, m - 1);
        }
        return Binarysearch(arr, target, m + 1, e);
    }
    
}

File: 0704-binary-search/0704-binary-search.cpp
class Solution {
public:
    
    int bs(vector<int>& arr, int n, int target){
        int low = 0;
        int high = n-1;
        while(low<=high){
            int mid = (low+high)/2;
            if(arr[mid] == target) return mid;
            if(arr[mid]>target){
                high = mid-1;
            }else{
                low = mid+1;
            }
        }
        return -1;
    }
    
    int search(vector<int>& nums, int target) {
        return bs(nums, nums.size(), target);
    }
};